### PR TITLE
add option to number counter-examples from zero

### PR DESCRIPTION
**Pull request by Ptival**
@atomb 

Here is a proposal implementation of the wanted feature, using a command-line flag named `--zero-based-cex` to alter the numbering scheme.

### CODE REVIEW

File: src/Main.hs [hunk 1]
```diff
@@ -176,6 +182,7 @@ options = OptSpec
     , timeout = Nothing
     , produceXml = False
     , printVersion = False
+    , zeroBasedCex = False
```
Comment (atomb) [inline src/Main.hs:185]: I think it'll help with integration if either the default is `True` or if the `--xml` flag sets it to true.

**Review by atomb — CHANGES_REQUESTED** [review]
I think this looks good, but have one minor change: for the XML output to be Kind 2 compatible, I think the `--xml` option should enable zero-based counter-examples, too.

**Comment by Ptival** [comment]
Well, this made me realize that the XML output is not the same at all as the terminal output.

New commit makes it so that both outputs obey the `zeroBasedCex` flag, and so that `--xml` implies `--zero-based-cex`.

There's some pretty ugly plumbing of the flag throughout the XML methods, maybe it does not warrant a proper record/datatype/reader yet...

**Review by atomb — APPROVED** [review]

**Comment by atomb** [comment]
It could be that we don't need the flag in the end, and that the XML output could always be zero-based, but I think this does the trick.

**Pull request merged**